runtime/gc_vlhgc/CycleStateVLHGC.hpp

```diff
@@ -40,11 +40,18 @@ class MM_CycleStateVLHGC : public MM_CycleState {
 	MM_VLHGCCycleStats _vlhgcCycleStats; /**< Stats for the various phases / operations within a cycle */
 	MM_SchedulingDelegate *_schedulingDelegate;
 
+	uintptr_t _desiredCompactWork; /**< Stats for desired amount of work to be compacted during a particular PGC cycle */
+	bool _useSlidingCompactor; /**< Stats to indicate If we surpassed survivor free memory. If so it's set to true, false otherwise */
+	bool _abortFlagRaisedDuringPGC; /**< Stats that indicate if PGC cycle aborted or not */
+
 	MM_CycleStateVLHGC()
 		: MM_CycleState()
 		, _vlhgcIncrementStats()
 		, _vlhgcCycleStats()
 		, _schedulingDelegate(NULL)
+		, _desiredCompactWork(0)
+		, _useSlidingCompactor(false)
+		, _abortFlagRaisedDuringPGC(false)
```

Comment:
While touching this let's _try_ to do some more cleanup.... There are a few of fields in base MM_CycleState that are VLHGC specific, that should move here.

It could be painful though,
1) base one is in OMR, so it will need more pull requests.
2) since it will move to a more speicific subclass, most sites that use those fields will have to upcast. Some of those sites will be outside of files that you've already touched

This is not a must - if it's too much of a mess, just forget it.

Reply:
Would those include the following?
```
MM_WorkPacketsVLHGC* _workPackets;
MM_MarkMap* _markMap; /**< Mark map central to activity of the current cycle */
uintptr_t _currentIncrement; /**< The index of the current increment within the cycle, starting at 0 (used for event reporting) */
bool _shouldRunCopyForward; /**< True if this cycle is to run a copy-forward-based attempt at reclaiming memory, false if mark-compact is to be used */
```
Did I miss something?

Reply:
Yeah, there is more - probably 2/3 of them are VLHGC specific. But you can be selective, and move only those that cause less pain.

Reply:
Ah I see, cool I think it'll be possible to move quite a few of these. I'll have a look.

Reply:
For example, _markDelegateState and _reasonForMarkCompactPGC are somewhat ok, while _collectionType and _markMap are not, since they appear in many more files that you have not touched.

Reply:
While you can give it a try locally, please don't push anything. I think we can just cut it here and deliver this change, and consider cleanup with separate change - they won't be small. I'll run some test now...

Reply:
I had a quick look and I was going to suggest the same :)